frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.tsx

```diff
@@ -180,11 +180,16 @@ const Header = ({
   <HeaderContainer>
     <HeaderClickable onClick={onBack}>
       {onBack && <Icon name="chevronleft" size={18} />}
-      <HeaderDatabaseName>{selectedDatabase.name}</HeaderDatabaseName>
+      <HeaderDatabaseName data-testid="source-database">
+        {selectedDatabase.name}
+      </HeaderDatabaseName>
     </HeaderClickable>
 
     {selectedSchema?.name && schemas.length > 1 && (
-      <HeaderSchemaName>- {selectedSchema.displayName()}</HeaderSchemaName>
+      <HeaderSchemaName>
+        {"- "}
+        <span data-testid="source-schema">{selectedSchema.displayName()}</span>
+      </HeaderSchemaName>
```

Comment:
I could've let this one go, but it really bothered me that we're asserting that the following string exists in the DOM "- schemaName". This just felt wrong.

This is not the ideal solution either and this whole component could be improved, but it does the job for now, IMO.
I'll follow up with another PR that fixes https://github.com/metabase/metabase/issues/39999